libraries/Update/src/HttpsOTAUpdate.cpp

```diff
@@ -0,0 +1,85 @@
+/* Cloud task
+
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
+#include <stdio.h>
+#include <string.h>
+
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "esp32-hal-log.h"
+#include "esp_http_client.h"
+#include "esp_https_ota.h"
+
+#include"HttpsOTAUpdate.h"
+#include"Esp.h"
+
+static TaskHandle_t ota_task_handle = NULL;
+const char * server_cert;
+const char * ota_url;
+bool skip_cert_common_name_check;
+void (*http_func_ptr)(HttpEvent_t*);
+bool success = false;
+bool fail = false;
+
+esp_err_t _http_event_handler(esp_http_client_event_t *evt)
+{
+    http_func_ptr(evt); 
+    return ESP_OK;
+}
+
+void aws_iot_task(void *param)
+{
+    esp_http_client_config_t config;
+    config.url = ota_url;
+    config.cert_pem = server_cert;
+    config.skip_cert_common_name_check = skip_cert_common_name_check;
+    config.event_handler = _http_event_handler;
+    esp_err_t ret = esp_https_ota(&config);
+    if(ret == ESP_OK){
+        success = true;
+    } else {
+        fail = true;
+        if(ota_task_handle != NULL) {
+            vTaskDelete(ota_task_handle);
+        }
+    }
+    while (1) {
+        if(fail == true)
+            break;
+        vTaskDelay(1000 / portTICK_PERIOD_MS);
+    }
+}
+
+bool HttpsOTAUpdateClass::ota_successful()
+{
+    return success;
+}
+
+bool HttpsOTAUpdateClass::ota_fail()
+{
+    return fail;
+}
+
+void HttpsOTAUpdateClass::onHttpEvent(void (*ptr)(HttpEvent_t*))
+{
+    http_func_ptr = ptr;
+}
+
+void HttpsOTAUpdateClass::start_firmware_upgrade(const char *cert, const char *url, bool skip_cert_cn_check)
+{
+    server_cert = cert;
+    ota_url = url;
+    skip_cert_common_name_check = skip_cert_cn_check;
+
+    if (xTaskCreate(&aws_iot_task, "aws_iot_task", 9216, NULL, 5, &ota_task_handle) != pdPASS) {
```

Comment:
Can you please use a macro for stack size, for better readability?

Reply:
Also, where is `ota_task_handle` assigned? Am I missing anything here?

Reply:
Oops, my bad. I confused the last parameter for the `void *pvParameters` (3rd last parameter)